python/lsst/ts/ess/common/socket_server.py

```diff
@@ -147,15 +149,19 @@ async def read_loop(self) -> None:
                         assert self.command_handler is not None
                         await self.command_handler.handle_command(cmd, **kwargs)
 
-                except asyncio.IncompleteReadError:
-                    self.log.exception("Read error encountered. Retrying.")
-
-        except asyncio.CancelledError:
-            self.log.debug("read_loop cancelled.")
         except Exception:
-            self.log.exception("read_loop failed.")
+            self.log.exception("read_loop failed. Disconnecting.")
+            self.log.debug("Stoppping sending telemetry.")
+            assert self.command_handler is not None
```

Comment:
I strongly suggest not putting asserts or raising exceptions in exception handlers (unless you are re-raising the original after doing some processing).

What should happen if the command handler is None? Why would you fail to execute the next two steps in that situation?

More generally I am leery of `assert` in production code (as opposed to unit tests) because the resulting error can be difficult to understand. It's usually better to raise the appropriate exception with good explanatory message (though takes an extra line of code).

Reply:
I added the asserts because MyPy complains otherwise that a NoneType doesn't have a certain method or property because `command_handler` is `Optional`. I changed the assert to an if so no exception is raised.